src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java

```diff
@@ -61,6 +63,36 @@ public List<User> getUsers(@RequestParam(value = "institution", required = false
         return institutionKey != null ? getByInstitution(institutionKey) : userService.findAll();
     }
 
+    @PreAuthorize("hasRole('" + SecurityConstants.ROLE_ADMIN + "') or #username == authentication.name")
+    @PutMapping(value = "/{username}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateUserInstitution(@PathVariable("username") String username, @RequestBody User user,
+                                      @RequestParam(value = "email", defaultValue = "true") boolean sendEmail) {
+        if (!username.equals(user.getUsername())) {
+            throw new BadRequestException("The passed user's username is different from the specified one.");
+        }
+        final User original = getByUsername(username);
+
+        assert original != null;
+
+        // validate institution update is valid
+        List<URI> institutionURIs = Arrays.asList(user.getInstitution(), original.getInstitution())
```

Comment:
i guess we do not need this

Reply:
@blcham removed in commit [3402342](https://github.com/kbss-cvut/record-manager/pull/35/commits/3402342481154547dc4c4f81c228f92d513013ec)